### PR TITLE
🎉 Explorers in the Data Catalog

### PR DESCRIPTION
**Pull request by ikesau**
This became a _much_ larger refactor than I thought it would, but I couldn't in good conscience add a bunch of forks to the already large `indexExplorerViewsToAlgolia` script without composing things a little more sensibly.

## Changes
- Separates record generating functions and the indexing scripts into multiple files (e.g. `baker/algolia/utils/charts.ts` and `baker/algolia/indexChartsToAlgola.ts`)
- Refactored Explorer View indexing to add available entities, and support for CSV-based and Grapher-based charts
  - The best way to make sense of this is to start at the bottom of `explorerViews.ts` and go up from there
  - In summary:  Get basic explorer metadata from the DB, get extra metadata from the GitCMS, prepare a "base record" for each view, enrich each base record depending on what type of view it is (indicator/grapher/csv, finalize them with a few extra processing steps
- Adds a new `indexExplorerViewsAndCharts` script/index which uses the `ChartHit` interface (plus a new "type" enum)
  - Explorer view scores are scaled to the same range as chart scores
- Adds support for explorer views in the data catalog
  - This probably needs a little more love now - I haven't gone over the code again since I wrote it (I think the rest of the code is ready to review, if you're feeling keen, Marcel)

### CODE REVIEW

File: baker/algolia/indexChartsToAlgolia.ts [hunk 1]
```diff
@@ -1,219 +1,9 @@
...
-        const score = computeScore(record)
-        records.push({ ...record, score })
-    }
-
-    return records
-}
+import { getChartsRecords } from "./utils/charts.js"
 
 const indexChartsToAlgolia = async () => {
     if (!ALGOLIA_INDEXING) return
```
Comment (marcelgerber) [inline baker/algolia/indexChartsToAlgolia.ts:9]: Do we want to start a Bugsnag session here like we do for `indexExplorerViews`?

File: baker/algolia/utils/explorerViews.ts
Comment (marcelgerber) [inline baker/algolia/utils/explorerViews.ts:1]: Wow, this is really nicely done! And, also, a lot of effort!

**Review by marcelgerber — APPROVED** [review]
Really nice work here, ike!

I left a few small comments here and there, but overall this seems great.

File: baker/algolia/utils/explorerViews.ts [hunk 2]
```diff
@@ -0,0 +1,759 @@
...
+        views_7d: e.explorerViews_7d,
+        viewTitleIndexWithinExplorer: e.viewTitleIndexWithinExplorer,
+        score: e.score,
+    }
+}
+
+/**
+ * Scale explorer scores to the range of grapher scores
+ * e.g. if the highest explorer score is 100 and the highest grapher score is 1000,
+ * we want to scale the explorer scores to be between 0 and 1000
```
[the remark this answers is not in the document]
Comment (ikesau) [reply baker/algolia/utils/explorerViews.ts:79]: Oh duh, yeah, that's way better!

**Review by marcelgerber — APPROVED** [review]
Good now, except for that you're getting an import error

**Pull request merged**